typings/styled-components.d.ts

```diff
@@ -35,14 +35,15 @@ type Attrs<P, A extends Partial<P>, T> = {
 export interface StyledComponentClass<P, T, O = P> extends ComponentClass<ThemedOuterStyledProps<O, T>> {
   extend: ThemedStyledFunction<P, T, O>;
 
-  withComponent<K extends keyof JSX.IntrinsicElements>(tag: K): StyledComponentClass<JSX.IntrinsicElements[K], T, O>;
-  withComponent(element: ComponentClass<P>): StyledComponentClass<P, T, O>;
+  withComponent<K extends keyof JSX.IntrinsicElements>(tag: K): StyledComponentClass<JSX.IntrinsicElements[K], T, JSX.IntrinsicElements[K]>;
+  withComponent<U extends Partial<P> = {}>(element: Component<U>): StyledComponentClass<U & P, T, O>
```

Comment:
The type of `element` here is incorrect (`Component` refers to a component _instance_, not a component class or function). It should be `element: ComponentType<U>`

Reply:
Ops, true.

Reply:
@mbrowne My bad, here I am using the local type `Component` from `styled-component` itself which is correct (not a instance):
```
type Component<P> = ComponentClass<P> | StatelessComponent<P>;
```